berserk/clients/users.py

```diff
@@ -55,6 +55,30 @@ def get_all_top_10(self) -> Dict[str, Any]:
         path = "/api/player"
         return self._r.get(path, fmt=LIJSON)
 
+    def get_player_by_autocomplete(
+        self, term: str, object: bool = False
+    ) -> Union[List[str], Dict[str, Any]]:
```

Comment:
* Use the `|` operator instead of Union, for consistency.
* We try to better type new response using typed dicts, and tests, see how it’s done in https://github.com/lichess-org/berserk/pull/38
* `term` is not descriptive even though it’s the named used in the api. Something like `partial_username` would be better.

Reply:
Hi @kraktus thank you for the quick response, i have fixed a few and there're still something I want to clarify my understanding.
It is suggested to create a class extending TypedDict and specify the field of its key value pair, and introduce it to check the return type of my API.

Reply:
1. Yes, like https://github.com/lichess-org/berserk/blob/50d7882f968d67629b2b952455e930f85f868947/berserk/types/account.py#L16
Maybe the returned info already match an existing dict
2. I’ll check on my side, don’t worry about it for now

Reply:
2. has been changed server-side, it nows returns 400 if too short.